### PR TITLE
Adding Unicycler to the workflow

### LINKED ISSUE #11: Circularizing plasmid assemblies
In the original code of runMA, assembling circularized plasmids was implemented and has yet to migrate over to YEAT.

Before this feature can be implemented, a few things need to be done.

1. Check and see if the circularized plasmids code is working.
2. Understand what it does.
3. Maybe look into various other assemblers that focus primarily on bacterial genome circularization?

### PR DESCRIPTION
**Pull request by danejo3**
The purpose of this PR is integrate `Unicycler`.

`Unicycler` is known for its hybrid assembly pipeline for bacterial genomes. When dealing with short reads, `unicycler` will use `spades` and _optimize_ it to:

- Try a wide range of k-mer sizes and automatically selects the best.
- Filter out low-depth parts of the assembly to remove contamination.
- Apply SPAdes repeat resolution to the graph (as opposed to disconnected contigs in a FASTA file).
- Reject low-confidence repeat resolution to reduce the rate of misassembly.
- Trim off graph overlaps so sequences aren't repeated where contigs join.

This PR will resolve #11 .

### CODE REVIEW

**Comment by danejo3** [comment]
As of right now, YEAT can only process illumina paired-end reads. In the future, I plan on adding additional support for unpaired reads and long reads.

Unicycler has support for all kinds of reads--paired-end, unpaired, and long. One thing that is unique to unicycler is its ability to do a hybrid assembly with both short and long reads. According to the unicycler documentation, the hybrid assembly will be the most accurate (which makes sense).

In this PR, I have provided support for unicycler but only paired end reads. In another PR, we'll need to add the support for all kinds of reads.

File: environment.yml
Comment (danejo3) [inline environment.yml:12]: I forgot to include `megahit` in the previous PR #13 .

**Comment by danejo3** [comment]
Another thing that is unique with `unicycler` is that it creates its own `Bandage` file called `assembly.gbk`. No support for bandage yet but, I think adding another snake rule for creating the files needed for `Bandage` would be great. For example, creating a `fastg` file from spades.

**Comment by danejo3** [comment]
Unicycler has some interesting options:
--min_fasta_length (default 100)
--mode (conservative/normal/bold) (conservative = less complete, less errors; bold = more complete, more errors)

File: yeat/tests/test_cli.py
Comment (standage) [inline yeat/tests/test_cli.py:65]: Just curious: what is the source of these new files?

**Review by standage — COMMENTED** [review]
No concerns with the code review. Once I get a chance to try it myself, I'll merge if I don't run into any issues.

File: yeat/tests/test_cli.py
Comment (danejo3) [reply yeat/tests/test_cli.py:65]: They are from `Unicycler`'s github. https://github.com/rrwick/Unicycler/tree/main/sample_data

These were some of the datasets that I found and Sean referred me to.

File: environment.yml
Comment (standage) [inline environment.yml:18]: I get the following when I try to set up a yeat environment from scratch on macOS.

```
$ mamba env update -f environment.yml 
conda-forge/osx-64                                          Using cache
conda-forge/noarch                                          Using cache
bioconda/osx-64                                             Using cache
bioconda/noarch                                             Using cache
pkgs/main/osx-64                                              No change
pkgs/main/noarch                                              No change
pkgs/r/noarch                                                 No change
pkgs/r/osx-64                                                 No change


Looking for: ['black=21.10b0', "fastp[version='>=0.23']", "fastqc[version='>=0.11']", "gzip[version='>=1.7']", "mash[version='>=2.3']", "megahit[version='>=1.2']", "pytest-cov[version='>=3.0']", "quast[version='>=5.0']", "seqtk[version='>=1.3']", "snakemake[version='>=6.10']", "spades[version='>=3.15']", "unicycler[version='>=0.5']", "versioneer[version='>=0.21']"]


Encountered problems while solving:
  - nothing provides requested unicycler >=0.5
```

Have you only tested on WSL, did you use an earlier version on macOS, did you install from source, or something else?

Comment (danejo3) [reply environment.yml]: Sounds good. I have dropped the `unicycler` version number to 0.4 and is working now.

**Review by standage — APPROVED** [review]
Slipped through the cracks for a few days. Looks good, thanks!

**Pull request merged**
